bash_archiving.sh

```diff
@@ -9,7 +9,24 @@ then
 	then
 		for (( index=1; index <= $countUSERS; index++))
 		do
-			word=$(sed 's/\(^[a-zA-Z0-9]*\).*/\1/g' /etc/passwd | tr '\n' ' ' | cut -d " " -f $index)
+			word="$(
+				awk -v I=$index '
+					BEGIN{
+						FS=":"
+						B=1
+					}
+
+					{
+						if($1~/^[[:alnum:]]+$/){
+							A[B++]=$1
+						}
```

Comment:
This is effectively doing some of what `sed` was doing, only the regular expression is cleaner, making use of a character class.

If the first field of the current line is equal to the regular expression of only any number of alphanumeric characters, then the associative array `A` gets a new index (the name of which is numeric, hence `B`) assigned the value of matching, aforementioned field.